query_server/query/src/extension/logical/optimizer_rule/implicit_type_conversion.rs

```diff
@@ -138,41 +136,43 @@ impl OptimizerRule for ImplicitTypeConversion {
     }
 }
 
-struct DataTypeRewriter<'a> {
+struct DataTypeRewriter {
     /// input schemas
-    schemas: Vec<&'a DFSchemaRef>,
+    schema: DFSchemaRef,
 }
 
-impl<'a> DataTypeRewriter<'a> {
+impl DataTypeRewriter {
+    pub fn new(schemas: Vec<&DFSchemaRef>) -> Self {
+        let mut schema = DFSchema::empty();
+        schemas.into_iter().for_each(|s| schema.merge(s.as_ref()));
```

Comment:
what is the purpose of merge

Reply:
reference here:

https://github.com/apache/arrow-datafusion/blob/4c7833ebfdb2d022830bb97862e0ce36b0b3d6b1/datafusion/optimizer/src/analyzer/type_coercion.rs#L70-L99

Reply:
i want to hear your thoughts